source/recipes/troubleshooting.md

```diff
@@ -0,0 +1,40 @@
+# Troubleshooting
+
+## What to do if a binary cache is down or unreachable?
+
+Pass `--option substitute false` to Nix commands.
+
+### How do I fix: error: querying path in database: database disk image is malformed
+
+Try:
+
+```shell-session
+$ sqlite3 /nix/var/nix/db/db.sqlite "pragma integrity_check"
+```
+
+Which will print the errors in the database. If the errors are due to missing
+references, the following may work:
+
+```shell-session
+$ mv /nix/var/nix/db/db.sqlite /nix/var/nix/db/db.sqlite-bkp
+$ sqlite3 /nix/var/nix/db/db.sqlite-bkp ".dump" | sqlite3 /nix/var/nix/db/db.sqlite
+```
+
+### How do I fix: error: current Nix store schema is version 10, but I only support 7
+
+This means you have upgraded Nix sqlite schema to a newer version, but then tried
+to use older Nix.
+
+The solution is to dump the db and use old Nix version to initialize it:
+
+```shell-session
+$ /path/to/nix/unstable/bin/nix-store --dump-db > /tmp/db.dump
+$ mv /nix/var/nix/db /nix/var/nix/db.toonew
+$ mkdir /nix/var/nix/db
+$ nix-store --init # this is the old nix-store
+$ nix-store --load-db < /tmp/db.dump
+```
+
+### I'm getting: writing to file: Connection reset by peer
+
+Too big files in src, out of resources (HDD space, memory)
```

Comment:
This is missing a suggestion on what to do.

```suggestion

This may mean you are trying to import a too large file or directory into the [Nix store](https://nixos.org/manual/nix/stable/glossary#gloss-store), or your machine is running out of resources, such as disk space or memory.

Try to reduce the size of the directory to import, or run [garbage collection](https://nixos.org/manual/nix/stable/command-ref/nix-collect-garbage).
```